enclave_manager/avalon_enclave_manager/avalon_enclave_info.py

```diff
@@ -0,0 +1,339 @@
+# Copyright 2018 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import json
+import time
+import random
+
+from ssl import SSLError
+from requests.exceptions import Timeout
+from requests.exceptions import HTTPError
+import avalon_crypto_utils.keys as keys
+import avalon_enclave_manager.ias_client as ias_client
+import avalon_enclave_manager.avalon_enclave as enclave
+
+import logging
+logger = logging.getLogger(__name__)
+
+TCF_HOME = os.environ.get("TCF_HOME", "../../../")
+SIG_RL_UPDATE_PERIOD = 8 * 60 * 60  # in seconds every 8 hours
+
+
+class EnclaveInfo():
+    """
+    Class to initialize enclave, signup enclave and hold
+    data obtained post signup - nonce, sealed_data,
+    verification_key, encryption_key, encryption_key_signature
+    proof_data, enclave_id.
+    """
+
+    __ias = None
+    __epid_group = None
+    __tcf_enclave_info = None
+    __sig_rl_update_time = None
+
+    # -------------------------------------------------------
+    def __init__(self, config):
+
+        # Initialize the keys that can be used later to
+        # register the enclave
+
+        enclave._SetLogger(logger)
+
+        self.initialize_enclave(config)
+        enclave_info = self.create_enclave_signup_data()
+        try:
+            self.nonce = enclave_info['nonce']
+            self.sealed_data = enclave_info['sealed_data']
+            self.verifying_key = enclave_info['verifying_key']
+            self.encryption_key = enclave_info['encryption_key']
+            self.encryption_key_signature = \
+                enclave_info['encryption_key_signature']
+            self.proof_data = enclave_info['proof_data']
+            self.enclave_id = enclave_info['enclave_id']
+        except KeyError as ke:
+            raise Exception("missing enclave initialization parameter; {}"
+                            .format(str(ke)))
+
+        self.enclave_keys = \
+            keys.EnclaveKeys(self.verifying_key, self.encryption_key)
+
+    # -------------------------------------------------------
+
+    def create_enclave_signup_data(self):
```

Comment:
Looks like this function is not used outside this class. If so, this is a private function and as a convention please rename this to `__create_enclave_signup_data()`. Similarly pre-pend single/double underscore for other functions in this class which can be private member functions. 
https://www.python.org/dev/peps/pep-0008/#method-names-and-instance-variables